src/ui/inventory.js

```diff
@@ -0,0 +1,182 @@
+import { CONSTANTS, FONTS } from "../constants/constants.js";
+import { getItemClass } from "./itemTypes.js";
+import { ITEMS } from "../constants/items.js";
+import { MATERIALS } from "../constants/materials.js";
+
+export class Inventory {
+    scene;
+    menu;
+    playerItems = []; // Pointer to cookies, store item name/material
+    inventory = []; // Images
+    curSelectedItemIndex = -1;
+
+    constructor(scene, inventory) {
+        this.scene = scene;
+        this.playerItems = inventory;
+
+        // Update and show inventory on startup
+        this.playerItems.forEach(item => {
+            // Add blanks
+            if (!Object.keys(item).length) {
+                this.addToInventory({});
```

Comment:
Why exactly do we need to fill it up with blanks?

Reply:
To maintain item locations on refresh. Currently I use the same function for populating on startup (where I know the correct index) and populating during run-time (where I don't know the correct index). 

I think an improvement I could do to resolve this would be to create two separate functions. One for inserting into inventory at a specific index and one that finds the first available index and calls the insert function on that index. Right now I have those two combined and so it requires me to make this additional operation. Can fix